presto-iceberg/src/test/java/io/prestosql/plugin/iceberg/TestIcebergMetadataListing.java

```diff
@@ -0,0 +1,197 @@
+                + ", _integer INTEGER"
+                + ", _double DOUBLE"
+                + ")";
+
+        assertQuerySucceeds(createSchema);
+        assertQuerySucceeds(createIcebergTable1);
+        assertQuerySucceeds(createIcebergTable2);
+        assertQuerySucceeds(createHiveTable);
+    }
+
+    @AfterClass
+    public void tearDown()
+    {
+        @Language("SQL") String dropSchema = "DROP SCHEMA "
+                + HIVE_CATALOG + "." + TEST_SCHEMA;
+
+        @Language("SQL") String dropIcebergTable1 = "DROP TABLE "
+                + ICEBERG_CATALOG + "." + TEST_SCHEMA + "." + ICEBERG_TABLE1;
+
+        @Language("SQL") String dropIcebergTable2 = "DROP TABLE "
+                + ICEBERG_CATALOG + "." + TEST_SCHEMA + "." + ICEBERG_TABLE2;
+
+        @Language("SQL") String dropHiveTable = "DROP TABLE "
+                + HIVE_CATALOG + "." + TEST_SCHEMA + "." + HIVE_TABLE;
+
+        assertQuerySucceeds(dropIcebergTable1);
+        assertQuerySucceeds(dropIcebergTable2);
+        assertQuerySucceeds(dropHiveTable);
+        assertQuerySucceeds(dropSchema);
+    }
+
+    @Test
+    public void testTableListing()
+    {
+        @Language("SQL") String showIcebergTables = "SHOW TABLES FROM "
+                + ICEBERG_CATALOG + "." + TEST_SCHEMA;
+
+        @Language("SQL") String expected = "VALUES "
+                + "'" + ICEBERG_TABLE1 + "'"
+                + ", '" + ICEBERG_TABLE2 + "'";
+
+        assertQuery(showIcebergTables, expected);
+    }
+
+    @Test
+    public void testTableColumnListing()
+    {
+        @Language("SQL") String countIcebergTableColumns = "SELECT count(*) FROM "
+                + ICEBERG_CATALOG + "." + INFORMATION_SCHEMA + "." + TABLE_COLUMNS.getTableName()
+                + " WHERE table_schema = " + "'" + TEST_SCHEMA + "'";
+
+        assertQuery(countIcebergTableColumns, "VALUES 6");
+    }
+
+    @Test
+    public void testTableDescribing()
+    {
+        @Language("SQL") String describeIcebergTable = "DESCRIBE "
+                + ICEBERG_CATALOG + "." + TEST_SCHEMA + "." + ICEBERG_TABLE2;
+
+        @Language("SQL") String expected = "VALUES "
+                + "('_string', 'varchar', '', '')"
+                + ", ('_integer', 'integer', '', '')"
+                + ", ('_double', 'double', '', '')"
+                + ", ('_boolean', 'boolean', '', '')";
+
+        assertQuery(describeIcebergTable, expected);
+    }
+
+    @Test
+    public void testTableValidation()
+    {
+        @Language("SQL") String selectIcebergTable1 = "SELECT * FROM "
+                + ICEBERG_CATALOG + "." + TEST_SCHEMA + "." + ICEBERG_TABLE1;
+
+        @Language("SQL") String selectHiveTable = "SELECT * FROM "
+                + ICEBERG_CATALOG + "." + TEST_SCHEMA + "." + HIVE_TABLE;
+
+        assertQuerySucceeds(selectIcebergTable1);
+        assertQueryFails(selectHiveTable, "Not an Iceberg table: .*");
+    }
+
+    private static DistributedQueryRunner createQueryRunner()
```

Comment:
Move this right after the ctor, to be consistent with other tests basing from ATQF